File: exercicios/01/IMC.cpp
// gcc -std=c++11 IMC.cpp -o saida

#include<iostream>
#include<string>
using namespace std;

class Pessoa{
    private:
        float imc, altura, peso;
        string nome;
    public:
        Pessoa();
        Pessoa(string n, float a, float p);
        ~Pessoa();
        void calcIMC();
        string toString();


};
Pessoa::Pessoa(){
    imc = 0;
    altura =0;
    peso=0;
    nome = "";
}
Pessoa::Pessoa(string n, float a, float p){
    nome = n;
    altura = a;
    peso =p;
}
Pessoa::~Pessoa(){
    cout<<"deletando "+ nome+"\n\n";
}
void Pessoa::calcIMC(){
    imc = peso/(altura *altura);
}
string Pessoa::toString(){
    string s = nome + " , seu imc é "+ to_string(imc);
    return s;
}


int main(){

    Pessoa * p ;
    string nome;
    float altura, peso;

    cout<<"\nSeu nome: ";
    getline(cin, nome);

    cout<<"\nSua altura: ";
    cin>>altura;

    cout<<"\nSeu peso: ";
    cin>>peso;

    p = new Pessoa(nome, altura, peso);
    p->calcIMC();

    cout<< "\n"+ p->toString()+ "\n\n";

    delete p;


    return 0;
}

File: exercicios/01-básico/01/IMC.java

public class IMC{

    String nome;
    double peso;
    double altura;
    double imc;

    public IMC(String nome, double peso, double altura){
        this.nome = nome;
        this.peso = peso;
        this.altura = altura;

        imc = calcIMC(peso, altura);
    }

    private double calcIMC(double peso, double altura){
          double imc = peso / ( altura * altura);
          return imc;
    }

    public double getIMC(){
        return imc;
    }

    public void setPeso(double peso){
        this.peso = peso;
        imc = calcIMC(this.peso,altura);
    }

    
    
    public String toString(){
        String str = nome +", seu peso é " + peso;
        return str;
    }
    public void mais10(){
        peso += 10;
        imc = calcIMC(peso,altura);
    }


    public static void main(String [] args){
    IMC gigante = new IMC("Gigante", 60, 1.75);


    System.out.println(gigante.toString());

    gigante.mais10();
    System.out.println(gigante.toString());

    gigante.mais10();
    System.out.println(gigante.toString());

    gigante.setPeso(76.0);
    System.out.println(gigante.toString());
    System.out.printf("Seu IMC é %.2f\n", gigante.getIMC());


    double imc =  gigante.getIMC() ;
        
    if( imc < 18.5 )
        System.out.println("Abaixo do peso");
    else if (imc >= 18.5 && imc <= 24.99)
        System.out.println("Regular peso");
    else if ( imc >= 25 && imc <= 29.99)
        System.out.println("Acima peso");
    else if ( imc >=  30 && imc <= 34.99)
        System.out.println("Obesidade I");
    else
        System.out.println("Obesidade II ou mais ");

        
    }

}



